### ISSUE TITLE
java.lang.ArithmeticException  com.journeyapps.barcodescanner.CameraPreview.m

### ISSUE THREAD

**Issue by teolemon** [journeyapps/zxing-android-embedded#409]
#### Description of the problem:

This is a bug report based on a crash reported by the Play Store. We call the library from a Open Data and Open Source barcode scanning application called Open Food Facts: 
https://play.google.com/store/apps/details?id=org.openfoodfacts.scanner

It seems this happens on Galaxy S8, on Android 8. I have no mean to reproduce. A quick search for an upstream bug didn't yield anything conclusive.

**Which library version are you using?** E.g. 3.6.0

**Which phone/tablet are you using, and which Android version does it run? (e.g. Samsung Galaxy S5,
  Android 5.0)** 
Report from the Play Store:
Android 8.0 | 3 | 100.0% 
Galaxy S8+ (dream2lte) | 2 | 66.7%
Galaxy S8 (dreamlte) | 1 | 33.3%

**Does the same happen on other devices or an emulator?** No.

**Can you reproduce the issue in the sample project included with the library? If not, can you
  provide your own sample project or sample code that produces this error?**

https://github.com/openfoodfacts/openfoodfacts-androidapp/blob/master/app/src/main/java/openfoodfacts/github/scrachx/openfood/views/ContinuousScanActivity.java

**In the case of an error do you have a stack trace or adb logs?**

```java.lang.ArithmeticException: 
  at com.journeyapps.barcodescanner.CameraPreview.m (CameraPreview.java:87)
  at com.journeyapps.barcodescanner.CameraPreview.b (CameraPreview.java:6)
  at com.journeyapps.barcodescanner.CameraPreview.access$300 (CameraPreview.java)
  or                     .access$400 (CameraPreview.java)
  or                     .previewSized (CameraPreview.java)
  or                     .rotationChanged (CameraPreview.java)
  at com.journeyapps.barcodescanner.CameraPreview$3.handleMessage (CameraPreview.java:12)
  at android.os.Handler.dispatchMessage (Handler.java:101)
  at android.os.Looper.loop (Looper.java:164)
  at android.app.ActivityThread.main (ActivityThread.java:6938)
  at java.lang.reflect.Method.invoke (Method.java)
  at com.android.internal.os.Zygote$MethodAndArgsCaller.run (Zygote.java:327)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1374)```

**Comment by dhruv0singh** [comment]
Description of the problem:
This bug is reported on the Splunk. I also have no mean to reproduce.

**Which library version are you using?** - 3.5.0

**Which phone/tablet are you using, and which Android version does it run?** (e.g. Samsung Galaxy S5,
Android 5.0)
Issue seems to be happening only on 8.0.0 OS Version. SAMSUNG NOTE 8 | SAMSUNG S8 PLUS | SAMSUNG S9 PLUS | SAMSUNG S7 EDGE

**Does the same happen on other devices or an emulator?** No.

**Can you reproduce the issue in the sample project included with the library? If not, can you
provide your own sample project or sample code that produces this error?**

--CAN'T--

In the case of an error do you have a stack trace or adb logs?


java.lang.ArithmeticException: divide by zero
--
_at com.journeyapps.barcodescanner.CameraPreview.void calculateFrames()(Unknown Source:98)
at com.journeyapps.barcodescanner.CameraPreview.void rotationChanged()(Unknown Source:6)
void previewSized(com.journeyapps.barcodescanner.Size)
void access$300(com.journeyapps.barcodescanner.CameraPreview,com.journeyapps.barcodescanner.Size)
com.journeyapps.barcodescanner.CameraPreview$StateListener access$400(com.journeyapps.barcodescanner.CameraPreview)
at com.journeyapps.barcodescanner.CameraPreview.void rotationChanged()(Unknown Source:0)
void previewSized(com.journeyapps.barcodescanner.Size)
void access$300(com.journeyapps.barcodescanner.CameraPreview,com.journeyapps.barcodescanner.Size)
com.journeyapps.barcodescanner.CameraPreview$StateListener access$400(com.journeyapps.barcodescanner.CameraPreview)
at com.journeyapps.barcodescanner.CameraPreview$3.boolean handleMessage(android.os.Message)(Unknown Source:12)
at android.os.Handler.dispatchMessage(Handler.java:101)
at android.os.Looper.loop(Looper.java:164)
at android.app.ActivityThread.main(ActivityThread.java:6938)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:327)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1374)_

**Comment by dhruv0singh** [comment]
Any update on this????????????????????

**Comment by brunojennrich** [comment]
unbelievable... this bug exists in 3.3.0 also. going back does not work.

zxing seems to be a more ans more irreliable tool

**Comment by kaushik1212** [comment]
I am facing the same issue. Any updates on this Issue please?

**Comment by rgroenewoudt** [comment]
I also have this issue. Isn't it enough to check if width/height is zero in calculateFrames() method?

**Comment by rkistner** [comment]
Should be fixed in v4.0.0.
